controller/linodemachine_controller.go

```diff
@@ -258,6 +246,19 @@ func (r *LinodeMachineReconciler) reconcile(
 		return
 	}
 
+	linodeClusterKey := client.ObjectKey{
+		Namespace: machineScope.LinodeMachine.Namespace,
+		Name:      machineScope.Cluster.Spec.InfrastructureRef.Name,
+	}
+
+	if err = r.Client.Get(ctx, linodeClusterKey, machineScope.LinodeCluster); err != nil {
+		if err = client.IgnoreNotFound(err); err != nil {
+			logger.Error(err, "Failed to fetch Linode cluster")
+		}
+
+		return ctrl.Result{}, err
```

Comment:
I hate being put in this position where I'm advocating for naked returns... but we're using naked returns in the majority of places in this function. @AshleyDumaine snuck one past me in Create but I think we should be consistent. 

I think that we need the named returns in order to handle the event creation on errors, so we should keep these returns in their birthday suits.

Reply:
Programmatically correct, because `if err = r.Client.Get` uses the original error, so we do not defer one error and return another.

Reply:
That is true, this is purely a style question. I can also see that it's not really particularly important I just think that using named returns is somewhat jarring anyway so I'd prefer that we keep it very consistent inside functions where we are using named returns. 

I would add that this is not a hill I wish to die on.

Reply:
> keep it very consistent

Totally agree. Empty returns are markers/hints that the function does something different. In our case, it enforces update of the status.

Reply:
changed the return to an empty return